aip/client-libraries/4235.md

```diff
@@ -38,14 +38,19 @@ If the aforementioned requirements are met for a given field, client library
 generators **must** enable automatic population of said field in the
 generated client.
 
+If the field value is not equal to default value at the time of sending the
```

Comment:
The field value can still be equal to the default value, but have been set by the user. I would reword this as something like:

> The field **must not** be automatically populated by the client if:
> - The field does not support explicit presence, and its value is not equal to the default
> - The field supports explicit presence, and presence indicates that it has been set (even to the default value)

Reply:
@jskeet: the Java implementation checks whether the field is null or empty when deciding whether to auto populate. Naive question - does your above statement imply that there is a potential edge case where the user explicitly sets the field to empty or null and thus the generator should *not* auto populate the field?

Reply:
Gotcha. I've updated the AIP to reflect the wording and may need to follow up separately re: the Java implementation as I don't believe that edge case is covered.